ES|QL mixed cluster tests failed on 8.17

ES|QL mixed cluster tests failed with `Unknown DataType for type name: date_nanos on 8.17`.
I can reproduce locally with `./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.15.4#javaRestTest" `

```
[2024-11-20T15:34:09,463][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [test-cluster-2] fatal error in thread [elasticsearch-error-rethrower], exiting
java.lang.AssertionError: java.io.IOException: Unknown DataType for type name: date_nanos
	at org.elasticsearch.transport.InboundHandler.handleRequest(InboundHandler.java:262) ~[elasticsearch-8.15.4.jar:?]
	at org.elasticsearch.transport.InboundHandler.messageReceived(InboundHandler.java:116) ~[elasticsearch-8.15.4.jar:?]
	at org.elasticsearch.transport.InboundHandler.inboundMessage(InboundHandler.java:97) ~[elasticsearch-8.15.4.jar:?]
	at org.elasticsearch.transport.TcpTransport.inboundMessage(TcpTransport.java:821) ~[elasticsearch-8.15.4.jar:?]
	at org.elasticsearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:124) ~[elasticsearch-8.15.4.jar:?]
	at org.elasticsearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:96) ~[elasticsearch-8.15.4.jar:?]
	at org.elasticsearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:61) ~[elasticsearch-8.15.4.jar:?]
	at org.elasticsearch.transport.netty4.Netty4MessageInboundHandler.channelRead(Netty4MessageInboundHandler.java:57) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[?:?]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[?:?]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[?:?]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[?:?]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) ~[?:?]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) ~[?:?]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689) ~[?:?]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652) ~[?:?]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[?:?]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[?:?]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[?:?]
	at java.lang.Thread.run(Thread.java:1570) ~[?:?]
```

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/union_types.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/union_types.csv-spec
@@ -718,6 +718,7 @@ count:long  |  @timestamp:date
 multiIndexTsNanosToDatetimeStats
 required_capability: union_types
 required_capability: union_types_remove_fields
+required_capability: to_date_nanos
 
 FROM sample_data, sample_data_ts_nanos
 | EVAL @timestamp = DATE_TRUNC(1 hour, TO_DATETIME(@timestamp))
@@ -754,6 +755,7 @@ multiIndexTsLongStatsDrop
 required_capability: union_types
 required_capability: union_types_agg_cast
 required_capability: casting_operator
+required_capability: to_date_nanos
 
 FROM sample_data, sample_data_ts_long, sample_data_ts_nanos
 | STATS count=count(*) BY @timestamp::datetime
@@ -774,6 +776,7 @@ multiIndexTsLongStatsInline2
 required_capability: union_types
 required_capability: union_types_agg_cast
 required_capability: casting_operator
+required_capability: to_date_nanos
 
 FROM sample_data, sample_data_ts_long, sample_data_ts_nanos
 | STATS count=count(*) BY @timestamp::datetime
@@ -917,6 +920,7 @@ multiIndexIpStringTsLong
 required_capability: union_types
 required_capability: metadata_fields
 required_capability: union_types_remove_fields
+required_capability: to_date_nanos
 
 FROM sample_data* METADATA _index
 | EVAL @timestamp = TO_DATETIME(@timestamp), client_ip = TO_IP(client_ip)
@@ -958,6 +962,7 @@ sample_data_ts_nanos | 2023-10-23T12:15:03.360Z  |  172.21.2.162  |  3450233    
 multiIndexIpStringTsLongDropped
 required_capability: union_types
 required_capability: metadata_fields
+required_capability: to_date_nanos
 
 FROM sample_data* METADATA _index
 | EVAL @timestamp = TO_DATETIME(@timestamp), client_ip = TO_IP(client_ip)
@@ -1000,6 +1005,7 @@ multiIndexIpStringTsLongRename
 required_capability: union_types
 required_capability: metadata_fields
 required_capability: union_types_remove_fields
+required_capability: to_date_nanos
 
 FROM sample_data* METADATA _index
 | EVAL ts = TO_DATETIME(@timestamp), host_ip = TO_IP(client_ip)
@@ -1041,6 +1047,7 @@ sample_data_ts_nanos | 2023-10-23T12:15:03.360Z  |  172.21.2.162  |  3450233    
 multiIndexIpStringTsLongRenameDropped
 required_capability: union_types
 required_capability: metadata_fields
+required_capability: to_date_nanos
 
 FROM sample_data* METADATA _index
 | EVAL ts = TO_DATETIME(@timestamp), host_ip = TO_IP(client_ip)
@@ -1083,6 +1090,7 @@ multiIndexIpStringTsLongRenameToString
 required_capability: union_types
 required_capability: metadata_fields
 required_capability: union_types_remove_fields
+required_capability: to_date_nanos
 
 FROM sample_data* METADATA _index
 | EVAL ts = TO_STRING(TO_DATETIME(@timestamp)), host_ip = TO_STRING(TO_IP(client_ip))
@@ -1125,6 +1133,7 @@ multiIndexWhereIpStringTsLong
 required_capability: union_types
 required_capability: metadata_fields
 required_capability: union_types_remove_fields
+required_capability: to_date_nanos
 
 FROM sample_data* METADATA _index
 | WHERE TO_LONG(@timestamp) < 1698068014937 AND TO_STRING(client_ip) == "172.21.2.162"
@@ -1141,6 +1150,7 @@ sample_data_ts_long  |  3450233              |  Connected to 10.1.0.3
 multiIndexWhereIpStringTsLongStats
 required_capability: union_types
 required_capability: union_types_remove_fields
+required_capability: to_date_nanos
 
 FROM sample_data*
 | WHERE TO_LONG(@timestamp) < 1698068014937 AND TO_STRING(client_ip) == "172.21.2.162"
@@ -1157,6 +1167,7 @@ multiIndexWhereIpStringLikeTsLong
 required_capability: union_types
 required_capability: metadata_fields
 required_capability: union_types_remove_fields
+required_capability: to_date_nanos
 
 FROM sample_data* METADATA _index
 | WHERE TO_LONG(@timestamp) < 1698068014937 AND TO_STRING(client_ip) LIKE "172.21.2.16?"
@@ -1173,6 +1184,7 @@ sample_data_ts_long  |  3450233              |  Connected to 10.1.0.3
 multiIndexWhereIpStringLikeTsLongStats
 required_capability: union_types
 required_capability: union_types_remove_fields
+required_capability: to_date_nanos
 
 FROM sample_data*
 | WHERE TO_LONG(@timestamp) < 1698068014937 AND TO_STRING(client_ip) LIKE "172.21.2.16?"
@@ -1189,6 +1201,7 @@ multiIndexMultiColumnTypesRename
 required_capability: union_types
 required_capability: metadata_fields
 required_capability: union_types_remove_fields
+required_capability: to_date_nanos
 
 FROM sample_data* METADATA _index
 | WHERE event_duration > 8000000
@@ -1207,6 +1220,7 @@ multiIndexMultiColumnTypesRenameAndKeep
 required_capability: union_types
 required_capability: metadata_fields
 required_capability: union_types_remove_fields
+required_capability: to_date_nanos
 
 FROM sample_data* METADATA _index
 | WHERE event_duration > 8000000
@@ -1226,6 +1240,7 @@ multiIndexMultiColumnTypesRenameAndDrop
 required_capability: union_types
 required_capability: metadata_fields
 required_capability: union_types_remove_fields
+required_capability: to_date_nanos
 
 FROM sample_data* METADATA _index
 | WHERE event_duration > 8000000

